src/backend/functions/change-requests.ts

```diff
@@ -110,12 +111,102 @@ const getChangeRequestByID: ApiRouteFunction = async (params: { id: string }) =>
   return buildSuccessResponse(changeRequestTransformer(requestedCR));
 };
 
+const createNewChangeRequest: ApiRouteFunction = async (_params, event) => {
+  if (event.body === undefined) {
+    return buildClientFailureResponse('No data for CR creation.');
+  }
+  const body = JSON.parse(event.body!);
+  if (
```

Comment:
If we are not interested in adding new libraries, then we might want to abstract each of these if statements into validation rules. Even if each function is only used once, it helps the main function be readable by distilling each if statement into a function with a descriptive name that says what it's doing.

We would turn:

`
if (
     body.submitterId === undefined ||
     body.wbsElementId === undefined ||
     body.type === undefined
   ) {
     return buildClientFailureResponse('Missing common CR field(s).');
   }`

Into:

`validateCommonCrFields(body)`

My main reason for this suggestion is that future methods could either use the same validation rules or at the very least, someone reading this function for the first time can quickly see several validation function calls and understand that each of those functions is checking a few fields.

Reply:
The last time we opted to abstract stuff ourselves rather than go for a library I regretted it cuz it ended up being a big headache and wasting a lot of time given how good and feature-rich React-Query turned out to be. We ended up switching to the package as soon as we hit the more complicated use case. I don't wanna make the same mistake here, but yeah I'm on the same page with abstraction.